### PR TITLE
Staging to master

### PR DESCRIPTION
**Pull request by miguelgfierro**
All the tests pass in staging

image

### CODE REVIEW

**Comment by roalexan** [comment]
Are the Windows tests "required"? If yes, there's some setting, I guess in devops, that needs to be updated. Not sure where off-hand.

File: reco_utils/common/tf_utils.py [hunk 1]
```diff
@@ -48,17 +48,39 @@ def pandas_input_fn(
         if isinstance(values[0], (list, np.ndarray)):
             values = np.array([l for l in values], dtype=np.float32)
         X[col] = values
-
-    input_fn = tf.estimator.inputs.numpy_input_fn(
+            
+    return lambda : _dataset(
```
Comment (gramhagen) [inline reco_utils/common/tf_utils.py]: unnecessary extra space after lambda

File: reco_utils/nni/nni_utils.py [hunk 2]
```diff
@@ -1,31 +1,32 @@
 import json
```
Comment (gramhagen) [inline reco_utils/nni/nni_utils.py]: missing license

File: reco_utils/recommender/lightgbm/lightgbm_utils.py [hunk 3]
```diff
@@ -8,32 +8,6 @@
 import collections
 import gc
```
Comment (gramhagen) [inline reco_utils/recommender/lightgbm/lightgbm_utils.py]: unused os import

File: reco_utils/recommender/lightgbm/lightgbm_utils.py [hunk 4]
```diff
@@ -82,60 +60,74 @@ def __init__(self, cate_cols, nume_cols, label_col, threshold=10, thresrate=0.99
...
+        logging.info("Target encoding cate features")
         # dynamic_targeting_encoding
         for item in tqdm(self.cate_cols):
             feats = df[item].values
             labels = df[self.label_name].values
-            feat_encoding = {'mean':[], 'count':[]}
-            feat_temp_result = collections.defaultdict(lambda : [0, 0])
-            self.save_cate_avgs[item] = collections.defaultdict(lambda : [0, 0])
+            feat_encoding = {"mean": [], "count": []}
+            feat_temp_result = collections.defaultdict(lambda: [0, 0])
```
Comment (gramhagen) [inline reco_utils/recommender/lightgbm/lightgbm_utils.py]: unused variable feat_temp_result

**Comment by miguelgfierro** [comment]
there was an error with nni, I added a bigger number in max retries

```
tests/integration/test_notebooks_python.py:162: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/anaconda/envs/nightly_reco_base/lib/python3.6/site-packages/papermill/execute.py:94: in execute_notebook
    raise_for_execution_errors(nb, output_path)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

nb = {'cells': [{'cell_type': 'code', 'metadata': {'inputHidden': True, 'hide_input': True}, 'execution_count': None, 'sour...d_time': '2019-05-24T12:24:18.694814', 'duration': 232.970518, 'exception': True}}, 'nbformat': 4, 'nbformat_minor': 2}
output_path = 'output.ipynb'

    def raise_for_execution_errors(nb, output_path):
        """Assigned parameters into the appropriate place in the input notebook
    
        Parameters
        ----------
        nb : NotebookNode
           Executable notebook object
        output_path : str
           Path to write executed notebook
        """
        error = None
        for cell in nb.cells:
            if cell.get("outputs") is None:
                continue
    
            for output in cell.outputs:
                if output.output_type == "error":
                    error = PapermillExecutionError(
                        exec_count=cell.execution_count,
                        source=cell.source,
                        ename=output.ename,
                        evalue=output.evalue,
                        traceback=output.traceback,
                    )
                    break
    
        if error:
            # Write notebook back out with the Error Message at the top of the Notebook.
            error_msg = ERROR_MESSAGE_TEMPLATE % str(error.exec_count)
            error_msg_cell = nbformat.v4.new_code_cell(
                source="%%html\n" + error_msg,
                outputs=[
                    nbformat.v4.new_output(output_type="display_data", data={"text/html": error_msg})
                ],
                metadata={"inputHidden": True, "hide_input": True},
            )
            nb.cells = [error_msg_cell] + nb.cells
            write_ipynb(nb, output_path)
>           raise error
E           papermill.exceptions.PapermillExecutionError: 
E           ---------------------------------------------------------------------------
E           Exception encountered at "In [14]":
E           ---------------------------------------------------------------------------
E           TimeoutError                              Traceback (most recent call last)
E           <ipython-input-14-c9381fa3d2ee> in <module>
E           ----> 1 check_metrics_written(wait=WAITING_TIME, max_retries=MAX_RETRIES)
E                 2 trials, best_metrics, best_params, best_trial_path = get_trials('maximize')
E           
E           /data/home/recocat/cicd/7/s/reco_utils/nni/nni_utils.py in check_metrics_written(wait, max_retries)
E                89         i += 1
E                90     if i == max_retries:
E           ---> 91         raise TimeoutError("check_metrics_written() timed out")
E                92 
E                93 
E           
E           TimeoutError: check_metrics_written() timed out

```

**Pull request merged**